### ISSUE TITLE
Login not working

### ISSUE THREAD

**Issue by MMore** [kviklet/kviklet#104]
Hey,

I wanted to quickly try your app but already the login fails because the path could not be found

<img width="473" alt="Screenshot 2024-05-12 at 18 00 28" src="https://github.com/kviklet/kviklet/assets/172760/b73f3638-e347-46cf-8207-3b08e357f981">

The regular login screen is shown but login fails. It's deployed via Fly and Docker image `ghcr.io/kviklet/kviklet:0.3.0`.

Any idea?

Apart from that it seems to be pretty rough still, but that's kinda expected with 0.3.0. Feels unusual to use a Java-based app for this btw.

**Comment by Askir** [comment]
Hmm that's strange.
Did you deploy it under the root path of your domain (kviklet-prod.internal in your case)? or did you add anything to the path?
Kivklet currently expects that the container is reachable directly on port 80 of the domain. So if you have some reverse proxying setup in between for subroutes like `http://kviklet-prod.internal/test` it wont work I believe.

Otherwise could you share a bit more about your deployment and especially networking setup? We have a demo deployment on AWS ECS which continues to work fine so I hope the release doesn't have a fundamental problem.
___
Regarding this
> Feels unusual to use a Java-based app for this btw.

We are using Kotlin to take advantage of the mature JDBC drivers for almost all databases, and because I am decently comfortable with the language of course.
A python or node app or even Rust would have also worked just fine but they don't have such a nice RDBMS abstraction as far as I know.

**Comment by VIDGuide** [comment]
Can also confirm running an instance on ECS via a subdomain, and not a sub-directory path and it's working perfectly fine for me.

**Comment by MMore** [comment]
Ah 80, the log stated 8080
> Tomcat initialized with port(s): 8080 (http)

That could be stated more clear in the README.

However, it seems, it's still not working for me. That's my `fly.toml`:
```toml
app = "kviklet-prod"

[build]
    image = "ghcr.io/kviklet/kviklet:0.3.0"

[env]
    # that will move to secrets soon
    SPRING_DATASOURCE_PASSWORD = "secret"
    SPRING_DATASOURCE_USERNAME = "user"
    SPRING_DATASOURCE_URL = "jdbc:postgresql://dbserver.flycast:5432/kviklet?sslmode=disable&currentSchema=public"

    INITIAL_USER_EMAIL = "bla@example.com"
    INITIAL_USER_PASSWORD= "someverysecurepassword"

[[services]]
  protocol = "tcp"
  internal_port = 80

  [[services.ports]]
    port = 80
    handlers = ["http"]
```

**Comment by Askir** [comment]
The 8080 is mapped internally by an nginx proxy. It's a bit of a hack to enable running the nginx and the tomcat server in the same container. But yes I'll clarify it in the readme.

Regarding the actual deployment... I never worked with fly.io but I've set up a trial account and set up a working deployment (I can at least log in) like this:
```toml
# fly.toml app configuration file generated for kviklet-test-crimson-glitter-6903 on 2024-05-13T10:30:04+08:00
#
# See https://fly.io/docs/reference/configuration/ for information about how to use this file.
#

app = 'kviklet-test-crimson-glitter-6903'
primary_region = 'arn'

[build]
  image = 'ghcr.io/kviklet/kviklet:0.3.0'

[env]
  INITIAL_USER_EMAIL = 'bla@example.com'
  INITIAL_USER_PASSWORD = 'someverysecurepassword'
  SPRING_DATASOURCE_PASSWORD = ''
  SPRING_DATASOURCE_URL = 'jdbc:h2:mem:mydb;MODE=LEGACY'
  SPRING_DATASOURCE_USERNAME = 'sa'
  SPRING_DATASOURCE_driverClassName = 'org.h2.Driver'

[http_service]
  internal_port = 80
  force_https = true
  auto_stop_machines = true
  auto_start_machines = true
  min_machines_running = 0
  processes = ['app']

[[services]]
  protocol = 'tcp'
  internal_port = 80

  [[services.ports]]
    port = 80
    handlers = ['http']

[[vm]]
  memory = '1gb'
  cpu_kind = 'shared'
  cpus = 1
```

I started with your config but after the interface opened the port mapping was nowhere to be found so I added 80 and they added the https handler to the config.

Note that this part:
```
SPRING_DATASOURCE_URL = 'jdbc:h2:mem:mydb;MODE=LEGACY'
SPRING_DATASOURCE_USERNAME = 'sa'
SPRING_DATASOURCE_driverClassName = 'org.h2.Driver'
```
is just to make spring use an in memory db instead of connecting to an actual postgres DB, however this will definitely cause issues because if you restart the container all info is lost. So I only did it because I didn't want to also create a postgres DB.

Does this help? Or is there a particular reason why you don't want to use the `http_service` configuration and use a pure service one instead?

**Comment by MMore** [comment]
Thanks for getting back to me!

I found out that it's not working without [Fly Proxy](https://fly.io/docs/networking/private-networking/#flycast-private-fly-proxy-services). So `kviklet-prod.internal` does not work while `kviklet-prod.fly.dev` works.
I did not have issues with others internal services, not sure what's different here 🤷‍♂️ 

We don't want to expose this service though, so publicly available Kviklet is not an option. Luckily we can use the private Fly Proxy via Flycast with `kviklet-prod.flycast`. That works too!
